tests/unit_tests/blas/level1/dot.cpp

```diff
@@ -103,7 +103,17 @@ int test(const device& dev, int N, int incx, int incy) {
     bool good;
     {
         auto result_accessor = result_buffer.template get_access<access::mode::read>();
-        good                 = check_equal(result_accessor[0], result_ref, N, std::cout);
+#ifndef ENABLE_CUBLAS_BACKEND
+        good = check_equal(result_accessor[0], result_ref, N, std::cout);
+#else
+        if (std::is_same<fp, fp_res>::value) {
+            good = check_equal(result_accessor[0], result_ref, N, std::cout);
+        }
+        else {
+            // Increase error margin for cuBLAS since computation is done in float instead of double.
+            good = check_equal(result_accessor[0], result_ref, N * 10e5, std::cout);
```

Comment:
Is it expected error for users in this case? Or should we add this information as known limitation?

Reply:
Good point. This is a known limitation since cuBLAS does not support this API. Before this PR, this test was not enabled, hence we did not see this limitation. Other option would be to throw not_supported exception. What do you think?

Reply:
I think if cuBLAS doesn't support this functionality we can throw not_supported exception, so that it will be transparent for users. I guess they also can work it around on their side, right?

Reply:
Yes, the user can call dot with double precision (slower but accurate)